### ISSUE TITLE
Isn't there a forward propagation in the train function?

### ISSUE THREAD

**Issue by Tankaishen** [dualslash/reptile-torch#1]
In this function, it seems that there is not a forward propagation.

def train(self, task):

    # Train from Scratch
    self.reset()

    # Outer Training Loop
    for outer_iteration in tqdm.tqdm(range(self.outer_iterations)):

      # Track Current Weights
      current_weights = deepcopy(self.model.state_dict())

      # Sample a new Subtask
      samples, task_theta = sample(task)

      # Inner Training Loop
      for inner_iteration in range(self.inner_batch_size):

        # Process Meta Learning Batches
        for batch in range(0, len(sample_space), self.meta_batch_size):

          # Get Permuted Batch from Sample
          perm = np.random.permutation(len(sample_space))
          idx = perm[batch: batch + self.meta_batch_size][:, None]

          # Calculate Batch Loss
          batch_loss = self.loss(sample_space[idx], samples[idx])
          batch_loss.backward()

          # Update Model Parameters
          for theta in self.model.parameters():

            # Get Parameter Gradient
            grad = theta.grad.data

            # Update Model Parameter
            theta.data -= self.inner_step_size * grad

          # Update Model Loss from Torch Model Tensor
          loss_tensor = batch_loss.cpu()
          self.current_loss += loss_tensor.data.numpy()
          self.current_batch += 1

      # Linear Cooling Schedule
      alpha = self.outer_step_size * (1 - outer_iteration / self.outer_iterations)

      # Get Current Candidate Weights
      candidate_weights = self.model.state_dict()

      # Transfer Candidate Weights to Model State Checkpoint
      state_dict = {candidate: (current_weights[candidate] + alpha * 
                               (candidate_weights[candidate] - current_weights[candidate])) 
                                for candidate in candidate_weights}
      self.model.load_state_dict(state_dict)
      
      # Log new Training Loss
      self.log.add_scalars('Model Estimate/Loss', 
                           {'Loss' : self.current_loss / self.current_batch}, 
                           outer_iteration)
